src/containers/AllCharacters.js

```diff
@@ -0,0 +1,52 @@
+import React, { PureComponent } from 'react';
+import Characters from '../components/Characters';
+import Loading from '../components/Loading';
+import { getCharacters } from '../services/rickAndMortyApi';
+
+
+export default class AllCharacters extends PureComponent {
+  state = {
+    characters: [],
+    loading: true,
+    currentPage: 1,
+    totalPages: 25
+  }
+
+  fetchCharacters = currentPage => {
+    this.setState({ loading: true });
+    getCharacters(currentPage)
+      .then(({ totalPages, characters }) => this.setState({ totalPages, characters, loading: false }));
+  }
+
+  componentDidMount() {
+    this.fetchCharacters(this.currentPage);
+  }
+
+  previousButton(currentPage) {
```

Comment:
Okay, I know I told you this is fine but after seeing it again and thinking a bit there is actually a better way to architect this to give the function a more specific meaning and to use the componentDidUpdate() lifecycle. I'd suggest writing a function like so for paging

```
  previousButton = () => {
     // use an arrow function to be sure state queues and updates correctly
     this.setState(state => {
       return { currentPage: state.currentPage + 1 };
    });
  }
```
and then within the componentDidUpdate() lifecycle...
```
componentDidUpdate(prevProps, prevState) {
     if(prevState.currentPage !== this.state.currentPage) {
         // do your fetch in here
     }
}
```
and then you can do a conditional in your button to disable or enable the button depending on what page you are at. 

Let me know if you have questions and I'll be happy to go over it with you.